src/tracker/inflights.rs

```diff
@@ -122,6 +127,19 @@ impl Inflights {
         self.count = 0;
         self.start = 0;
     }
+
+    /// Free unused memory
+    #[inline]
+    pub fn free_mem(&mut self) {
+        // remove unused buffer, from start + count to the end of buffer
+        // ignore first index to self.start - 1
+        let len = self.buffer.len();
+        for _ in self.start + self.count..len {
+            self.buffer.pop();
+        }
```

Comment:
We initialize the buffer with fixed capacity, so we need to remove element for reducing allocated memory, any better ways?

Reply:
How about don't touch it if it's not empty? So the resource can be freed later, and no allocations are introduced.

Reply:
> How about don't touch it if it's not empty? 

I am not sure how frequently the buffer is empty. If it happens a lot, this approach is good enough.